lib/Doctrine/DBAL/Connections/MasterSlaveConnection.php

```diff
@@ -85,7 +86,7 @@ class MasterSlaveConnection extends Connection
     /**
      * Master and slave connection (one of the randomly picked slaves).
      *
-     * @var \Doctrine\DBAL\Driver\Connection[]
+     * @var DriverConnection[]|null[]
```

Comment:
Isn't it something like `(DriverConnection|null)[]`? Not all values are necessary of the same type.

Reply:
Such notation is more correct, but not supported i.e. by IDEs (PhpStorm will break completely). Pretty much same like intersections vs. unions we discussed earlier somewhere else.